sdk/core/azure-core-amqp/inc/azure/core/amqp/models/amqp_header.hpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <azure/core/internal/unique_handle.hpp>
+#include <azure/core/nullable.hpp>
+#include <chrono>
+#include <cstddef>
+#include <cstdint>
+#include <exception>
+#include <stdexcept>
+#include <vector>
+
+struct HEADER_INSTANCE_TAG;
+
+template <> struct Azure::Core::_internal::UniqueHandleHelper<HEADER_INSTANCE_TAG>
+{
+  static void FreeAmqpHeader(HEADER_INSTANCE_TAG* obj);
+
+  using type = Azure::Core::_internal::BasicUniqueHandle<HEADER_INSTANCE_TAG, FreeAmqpHeader>;
+};
+
+namespace Azure { namespace Core { namespace Amqp { namespace Models {
+
+  using UniqueMessageHeaderHandle = Azure::Core::_internal::UniqueHandle<HEADER_INSTANCE_TAG>;
+
+  struct MessageHeader final
+  {
+
+    MessageHeader() = default;
+    ~MessageHeader() = default;
+
+    bool operator==(MessageHeader const&) const noexcept;
+
+    /** @brief True if the message is considered "durable"
+     *
+     * @remarks For more information, see [AMQP
+     * Section 3.2.1](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-header).
+     */
+    bool Durable{false};
+
+    /** @brief Priority of the message.
+     *
+     * @remarks For more information, see [AMQP
+     * Section 3.2.1](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-header).
+     */
+    std::uint8_t Priority{4}; // Default priority is 4.
```

Comment:
As I was reviewing these, I kept having to refer to the AMQP spec for even the basic understanding, for example does higher priority mean a larger number or lower.

Maybe some of those descriptions could add value if they were in-lined into the comment, including the fact that the default is 4. Why did we pick that btw and what are the implications of doing so?

What is the range of valid priorities? Do we validate those?

Also, as the reader, I immediately wonder what does durable mean?
> durable messages MUST NOT be lost even if an intermediary is unexpectedly terminated and restarted

Reply:
If you are implementing an AMQP client, the code expects that you understand the AMQP protocol and its options. The links are there to help people learn more.

We don't provide similar levels of details for HTTP (for instance, we don't provide detailed information on the exact semantics of every possible HTTP header). I don't know why we need to do more than refer to the appropriate AMQP protocol documentation, unless this is covering a specific part of our implementation.

That is especially true for structures which are completely defined within the AMQP specification, like the message header and properties.